### PR TITLE
Gate: fetchTicker, fetchTickers, add option support

### PR DESCRIPTION
**Pull request by Dan-krm**
Added option support to fetchTicker and fetchTickers:

### fetchTicker:
```
gate.fetchTicker (BTC/USDT:USDT-230609-26500-C)
2023-06-08T05:53:32.248Z iteration 0 passed in 232 ms

                      symbol | timestamp | datetime | high | low | bid | bidVolume | ask | askVolume | vwap | open | close | last | previousClose | change | percentage | average | baseVolume | quoteVolume
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
BTC/USDT:USDT-230609-26500-C |           |          |      |     | 128 |         1 | 186 |       -52 |      |      |   111 |  111 |               |        |            |         |            |
1 objects
```

### fetchTickers:
```
gate.fetchTickers (BTC/USDT:USDT-230609-26500-C)
2023-06-08T06:00:24.755Z iteration 0 passed in 412 ms

{
  'BTC/USDT:USDT-230609-26500-C': {
    symbol: 'BTC/USDT:USDT-230609-26500-C',
    timestamp: undefined,
    datetime: undefined,
    high: undefined,
    low: undefined,
    bid: 124,
    bidVolume: 67,
    ask: 183,
    askVolume: -61,
    vwap: undefined,
    open: undefined,
    close: 111,
    last: 111,
    previousClose: undefined,
    change: undefined,
    percentage: undefined,
    average: undefined,
    baseVolume: undefined,
    quoteVolume: undefined,
    info: {
      vega: '5.56112',
      leverage: '68.002877307275',
      ask_iv: '0.4214',
      delta: '0.40375',
      last_price: '111',
      theta: '-98.36021',
      bid1_price: '124',
      mark_iv: '0.3819',
      name: 'BTC_USDT-20230609-26500-C',
      bid_iv: '0.3156',
      ask1_price: '183',
      mark_price: '156.57',
      position_size: '26',
      bid1_size: '67',
      ask1_size: '-61',
      gamma: '0.0007'
    }
  }
}
```

### CODE REVIEW

File: ts/src/gate.ts [hunk 1]
```diff
@@ -2141,10 +2145,21 @@ export default class gate extends Exchange {
...
+        if (market['type'] === 'option') {
+            const marketId = market['id'];
+            const optionParts = marketId.split ('-');
+            request['underlying'] = this.safeString (optionParts, 0);
+        }
         const response = await this[method] (this.extend (request, query));
-        const ticker = this.safeValue (response, 0);
-        return this.parseTicker (ticker, market);
+        const result = [];
+        for (let i = 0; i < response.length; i++) {
```
Comment (carlosmiei) [inline ts/src/gate.ts:2157]: @Dan-krm fethTicker should return a `Ticker`, but doing this we're changing the return type 
image

File: ts/src/gate.ts [hunk 2]
```diff
@@ -2141,10 +2145,21 @@ export default class gate extends Exchange {
...
+            const entry = response[i];
+            const ticker = this.parseTicker (entry);
```
Comment (carlosmiei) [inline ts/src/gate.ts:2159]: also we are unnecessarily parsing hundreds of tickers when we just output one

**Comment by JamesMessick1965** [comment]
bc1q6k3yjaqzef65t6vhzsyn09ncz5vhc3q53pp36s

File: ts/src/gate.ts (hunk 1 above)
Comment (Dan-krm) [reply ts/src/gate.ts]: Fair enough I knew there were some issues with my approach here, I like what you did instead

**Pull request merged**